modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SearchIndexResourceImpl.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.search.experiences.rest.internal.resource.v1_0;
+
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.search.index.IndexInformation;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.search.experiences.rest.dto.v1_0.SearchIndex;
+import com.liferay.search.experiences.rest.resource.v1_0.SearchIndexResource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Brian Wing Shun Chan
+ */
+@Component(
+	enabled = false,
+	properties = "OSGI-INF/liferay/rest/v1_0/search-index.properties",
+	scope = ServiceScope.PROTOTYPE, service = SearchIndexResource.class
+)
+public class SearchIndexResourceImpl extends BaseSearchIndexResourceImpl {
+
+	public List<SearchIndex> getSearchIndexes() {
+		List<SearchIndex> searchIndexes = new ArrayList<>();
+
+		String companyIndexName = _indexInformation.getCompanyIndexName(
+			contextCompany.getCompanyId());
+
+		for (String indexName :
+				ArrayUtil.remove(
+					_indexInformation.getIndexNames(), companyIndexName)) {
+
+			if (!indexName.startsWith(companyIndexName)) {
+				continue;
+			}
+
+			searchIndexes.add(
+				new SearchIndex() {
+					{
+						name = indexName;
+						suffix = _getIndexName(companyIndexName, indexName);
+					}
+				});
+		}
+
+		return searchIndexes;
+	}
+
+	@Override
+	public Page<SearchIndex> getSearchIndexesPage() throws Exception {
+		return Page.of(getSearchIndexes());
+	}
+
+	private String _getIndexName(String companyIndexName, String indexName) {
+		String indexNamePrefix = companyIndexName.substring(
+			0,
+			companyIndexName.indexOf(
```

Comment:
i think substring() needs to be passed in something like `0, indexOf(...) + companyId.length` ?

Reply:
companyIndexName is something like "liferay-12345" which is a predefined format. I have a unit test to verify these.

Reply:
and liferay- is the index prefix that a user can configure it